### PR TITLE
c: fix parameter captures

### PR DESCRIPTION
**Pull request by vigoux**
Leveraging @theHamsta changes.

### CODE REVIEW

**Comment by theHamsta** [comment]
Doesn't this make also the `reference_declarator` captures obsolete (they work like `pointer_declarator`). C++ highlights/locals.

**Comment by theHamsta** [comment]
@vigoux can you update this PR for C++? The same change you did for `pointer_declarator` should apply for `address_declarator`

**Comment by vigoux** [comment]
@theHamsta I added the C++ parameter thing, do you think it's good now ?

File: queries/c/locals.scm [hunk 1]
```diff
@@ -8,8 +8,12 @@
...
 (pointer_declarator
   declarator: (identifier) @definition.var)
-(parameter_declaration
-  declarator: (identifier) @definition.parameter)
+; (parameter_declaration
+;   declarator: (identifier) @definition.parameter)
+
+((identifier) @definition.parameter
+  (#has-ancestor? @definition.parameter "parameter_declaration"))
+
```
Comment (theHamsta) [inline queries/c/locals.scm:16]: I guess removing is sufficient since this query already comes from C

**Review by theHamsta — APPROVED** [review]
I guess this is good to go! We have to be careful no to use #has-... In combination with identifier too often. There're really a lot of identifiers in a file and you have to check all parents till the root

File: queries/c/locals.scm (hunk 1 above)
Comment (theHamsta) [reply queries/c/locals.scm:16]: Wornf position. I mean in the copy file. I'm on mobile right now. Too Little screen to big fingers...

File: queries/cpp/highlights.scm [hunk 2]
```diff
@@ -19,6 +17,9 @@
 (optional_parameter_declaration
     declarator: (_) @parameter)
 
+((identifier) @parameter
+
+  (#has-ancestor? @parameter "parameter_declaration"))
```
Comment (theHamsta) [inline queries/cpp/highlights.scm:22]: Meant this one. It's already in C highlights

Comment (vigoux) [reply queries/cpp/highlights.scm:22]: Oh yeah sorry huh

File: queries/c/highlights.scm [hunk 3]
```diff
@@ -157,11 +157,8 @@
 (comment) @comment
 
 ;; Parameters
-(parameter_declaration
-  declarator: (identifier) @parameter)
-
-(parameter_declaration
-  declarator: (pointer_declarator) @parameter)
+((identifier) @parameter
+  (#has-ancestor? @parameter "parameter_declaration"))
```
Comment (theHamsta) [inline queries/c/highlights.scm:161]: We probably want to avoid `#has-ancestor` since checking the parent has become expensive.

**Review by theHamsta — CHANGES_REQUESTED** [review]

**Comment by theHamsta** [comment]
Closing since this PR is stale for a long time and we want to avoid `has-ancestor?`. But I oberved a similar problem today with references to variadic template arguments

**Pull request closed without merging**